src/modules/configs/schemas/application.py

```diff
@@ -0,0 +1,158 @@
+                            "type": "integer",
+                            "minimum": 1
+                        }
+                    },
+                    "additionalProperties": False,
+                    "required": ["critical"]
+                }
+            },
+            "additionalProperties": False,
+            "required": [
+                "type",
+                "root_path",
+                "endpoint",
+                "limits"
+            ]
+        },
+        "permissions": {
+            "type": "object",
+            "properties": {
+                "primary": {
+                    "type": "object",
+                    "properties": {
+                        "uid": {
+                            "type": "integer"
+                        },
+                        "gid": {
+                            "type": "integer"
+                        },
+                        "mode": {
+                            "type": "string",
+                            "pattern": "^0o[1-7][0-7]{2}$"
+                        }
+                    },
+                    "additionalProperties": False,
+                    "required": [
+                        "uid",
+                        "gid",
+                        "mode"
+                    ]
+                }
+            },
+            "additionalProperties": False,
+            "required": ["primary"]
+        },
+        "resources": {
+            "type": "object",
+            "properties": {
+                "thread_count": {
+                    "type": "integer"
+                }
+            },
+            "additionalProperties": False,
+            "required": ["thread_count"]
+        },
+        "workdir": {
+            "type": "string",
+            "maxLength": 4096
+        },
+        "naming": {
+            "type": "object",
+            "properties": {
+                "log_file": {
+                    "type": "string",
+                    "maxLength": 255
+                },
+                "codename": {
+                    "type": "string",
+                    "maxLength": 20,
+                    "minLength": 1
+                }
+            },
+            "additionalProperties": False,
+            "required": []
+        },
+        "notify": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "maxLength": 4096,
+                "minLength": 11
+            }
+        },
+        "exclude": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "maxLength": 4096,
+                "minLength": 6
+            }
+        },
+        "include": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "maxLength": 4096,
+                "minLength": 6
+            }
+        }
+    },
+    "additionalProperties": False,
+    "required": ["containers"]
```

Comment:
we need this because it's the exclusive rule
```
    "containers_sata_device_path": {
        "mode": "exclusive",
        "exclusive": ("application",)
    },
    "containers_sata_mount_path": {
        "mode": "exclusive",
        "exclusive": ("application",)
    },
```

Reply:
It's indirectly required... We could postpone validation of that key until final conf but it's inefficient. My bad, I believe I was thinking about adding those to some other conf.